blockchain/blockchain.go

```diff
@@ -184,10 +204,27 @@ func (bc *blockchain) CommitBlock(blockTime time.Time, blockHash, appHash []byte
 
 func (bc *blockchain) save() error {
 	if bc.db != nil {
-		encodedState, err := bc.Encode()
+		genesisJSON, err := bc.genesisDoc.JSONBytes()
+		if err != nil {
+			return err
+		}
+
+		validatorSetJSON, err := bc.validatorSet.JSONBytes()
+		if err != nil {
+			return err
+		}
+
+		u := map[string]string{}
```

Comment:
Yep as before I don't like this serialisation style much. I think the solution of explicitly defining the `PersistedState` is better, would rather that was reinstated.

Reply:
Let we discuss more about it. Maybe there is third way. `PersistedState` is used just for serializing/deserializing. We can back to `PersistedState`, but don't you think this bring more complexity to the code.

Reply:
It is possible to use an in-line anonymous struct for serialisation which would contain it to the method - however this misses out centralising our 'model of serialisation'. I think the implicit model of serialisation defined separately from the model of deserialisation is more complex.

A third way might be to separate our 'models of serialisation' from our 'models of computation' - it sounds to me like this kind of containment/separation might be what you are after. I wouldn't be against having serialisation models moved to the bottom of file, a separate file, or maybe even a sub-package if there were more of them (or if they were used elsewhere). This is comparable to what you might do with protobuf generated structs (also models of serialisation).  For which matter we may transition to at some point - though we would still have the structs.